src/main/java/crawlercommons/sitemaps/SiteMapParser.java

```diff
@@ -209,13 +209,30 @@ public AbstractSiteMap parseSiteMap(String contentType, byte[] content, URL url)
             if (XML_MEDIA_TYPES.contains(mediaType)) {
                 return processXml(url, content);
             } else if (TEXT_MEDIA_TYPES.contains(mediaType)) {
-                return processText(url.toString(), content);
+                return processText(url, content);
             } else if (GZ_MEDIA_TYPES.contains(mediaType)) {
-                return processGzip(url, content);
-            } else {
-                mediaType = MEDIA_TYPE_REGISTRY.getSupertype(mediaType); // Check parent
-                return parseSiteMap(mediaType.toString(), content, url);
+                InputStream decompressed;
+                MediaType embeddedType;
+                try {
+                    decompressed = new GZIPInputStream(new ByteArrayInputStream(content));
```

Comment:
A ByteArrayInputStream does not hold file handle which may leak, the stream objects are garbage collected anyway.